gobblin-utility/src/main/java/org/apache/gobblin/util/logs/LogCopier.java

```diff
@@ -271,7 +282,23 @@ void checkSrcLogFiles() throws IOException {
         LOGGER.error("Failed LogCopyTask - {}", e);
       }
     }
+    if (needToUpdateDestFs) {
+      if(destFsSupplier == null) {
+        throw new IOException("Try to update dest fileSystem but destFsSupplier has not been set, there might be something wrong in code");
+      }
+      this.destFs.close();
+      this.destFs = destFsSupplier.getFileSystem();
+      LOGGER.info("Dest fs updated" + destFs.toString());
 
+    }
+    if (needToUpdateSrcFs) {
```

Comment:
Does this boolean need to be reset back to false after the logic in the if {..} block has finished?